File: ASG1_2.pl
#! /usr/bin/perl

use strict;
use warnings;
use IO::File ;
use IO::Compress::Deflate qw(deflate $DeflateError) ;

# read file into file stream $input
my $input = new IO::File "<NazrulYusni.txt"
    or die "Cannot open 'NazrulYusni.txt': $!\n" ;

# convert input data into buffer
my $buffer ;
deflate $input => \$buffer
    or die "Convert failed: $DeflateError\n";

# set output file
my $output = new IO::File ">NazrulYusni.bin"
    or die "Cannot open bin file: $!\n";

# write converted data into output file
print $output $buffer;

# output and input files are automatically closed

File: ASG1_2.java
import java.util.zip.DeflaterOutputStream;
import java.util.zip.InflaterInputStream;
import static java.nio.file.StandardOpenOption.*;
import java.nio.file.*;
import java.io.*;
import java.nio.charset.Charset;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ASG1_2 {

  public static void main(String[] args) throws IOException  {

    String filein = "NazrulYusni.txt";
    String allwords = "";
    // Use this for reading the data.
    byte[] buffer = new byte[100];

    FileInputStream inputStream = new FileInputStream(filein);
    int total = 0;
    int nRead = 0;

    //Read file in bytes
    while((nRead = inputStream.read(buffer)) != -1) {
        // Convert buffer to String so we can collect it.
        String thisBuffer = new String(buffer);
        // Collect all string in buffer
        allwords = allwords + thisBuffer;
        total += nRead;
    }
    // Close files.
    inputStream.close();

    // Convert read string to binary bytes
    byte[] binallwords = compress(allwords);

    // Open file in Binary for write
    Path fileout = Paths.get("NazrulYusni.bin");
    //FileOutputStream outputStream = new FileOutputStream(fileout);

    OutputStream out = new BufferedOutputStream(Files.newOutputStream(fileout, CREATE));
    // write data line by line
    out.write(binallwords, 0, binallwords.length);
    // close fileout
    out.close();
    System.out.println("Binary files created");

  } // ----------------- End main ----------------





  public static byte[] compress(String text) {
      ByteArrayOutputStream baos = new ByteArrayOutputStream();
      try {
          OutputStream out = new DeflaterOutputStream(baos);
          out.write(text.getBytes("UTF-8"));
          out.close();
      } catch (IOException e) {
          throw new AssertionError(e);
      }
      return baos.toByteArray();
  }



}
